pkg/codegen/docs/gen_creation_example_for_yaml.go

```diff
@@ -0,0 +1,193 @@
+package docs
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+// collapseToken converts an exact token to a token more suitable for
+// display. For example, it converts
+//
+//	  fizz:index/buzz:Buzz => fizz:Buzz
+//	  fizz:mode/buzz:Buzz  => fizz:mode:Buzz
+//		 foo:index:Bar	      => foo:Bar
+//		 foo::Bar             => foo:Bar
+//		 fizz:mod:buzz        => fizz:mod:buzz
+func collapseResourceTokenForYAML(token string) string {
+	tokenParts := strings.Split(token, ":")
+
+	if len(tokenParts) == 3 {
+		title := func(s string) string {
+			r := []rune(s)
+			if len(r) == 0 {
+				return ""
+			}
+			return strings.ToTitle(string(r[0])) + string(r[1:])
+		}
+		if mod := strings.Split(tokenParts[1], "/"); len(mod) == 2 && title(mod[1]) == tokenParts[2] {
+			// aws:s3/bucket:Bucket => aws:s3:Bucket
+			// We recourse to handle the case foo:index/bar:Bar => foo:index:Bar
+			tokenParts = []string{tokenParts[0], mod[0], tokenParts[2]}
+		}
+
+		if tokenParts[1] == "index" || tokenParts[1] == "" {
+			// foo:index:Bar => foo:Bar
+			// or
+			// foo::Bar => foo:Bar
+			tokenParts = []string{tokenParts[0], tokenParts[2]}
+		}
+	}
+
+	return strings.Join(tokenParts, ":")
+}
+
+func genCreationExampleSyntaxYAML(r *schema.Resource) string {
+	indentSize := 0
+	buffer := bytes.Buffer{}
+	write := func(format string, args ...interface{}) {
+		buffer.WriteString(fmt.Sprintf(format, args...))
+	}
+
+	indent := func() {
+		buffer.WriteString(strings.Repeat(" ", indentSize))
+	}
+
+	indended := func(f func()) {
+		indentSize += 2
+		f()
+		indentSize -= 2
+	}
+
+	seenTypes := codegen.NewStringSet()
+	var writeValue func(valueType schema.Type)
+	writeValue = func(valueType schema.Type) {
+		switch valueType {
+		case schema.BoolType:
+			write("True|False")
+		case schema.IntType:
+			write("0")
+		case schema.NumberType:
+			write("0.0")
+		case schema.StringType:
+			write("\"string\"")
+		case schema.ArchiveType:
+			write("\n")
+			indended(func() {
+				indent()
+				write("Fn::FileAsset: ./file.txt")
+			})
+		case schema.AssetType:
+			write("\n")
+			indended(func() {
+				indent()
+				write("Fn::StringAsset: \"example content\"")
+			})
+		}
+
+		switch valueType := valueType.(type) {
+		case *schema.ArrayType:
```

Comment:
This doesn't seem to be generating "normal" YAML - I would have expected these to use the `-` syntax.  That is instead of this:

```yaml
name: example
runtime: yaml
resources:
  iamResource:
    type: example:myModule:IamResource
    properties:
      config: 
        auditLogConfigs: [
          exemptedMembers: ["string"]
          logType: LOG_TYPE_UNSPECIFIED|ADMIN_READ|DATA_WRITE|DATA_READ
        ]
        service: "string"
```

I would expect this:

```yaml
name: example
runtime: yaml
resources:
  iamResource:
    type: example:myModule:IamResource
    properties:
      config:
        auditLogConfigs:
        - exemptedMembers:
          - string
          logType: LOG_TYPE_UNSPECIFIED|ADMIN_READ|DATA_WRITE|DATA_READ
        service: string
```

Reply:
Will fix this, I though that form makes the shape more clear (for example for array of objects)